### PR TITLE
iteration 1

### PR DESCRIPTION
**Pull request by AlexSamarkin**

### CODE REVIEW

File: src/main/java/com/acme/dbo/txlog/ConsoleOutput.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.acme.dbo.txlog;
+
+public class ConsoleOutput {
+    public static <T> void sysOut(T message) {
```
Comment (eugene-krivosheyev) [inline src/main/java/com/acme/dbo/txlog/ConsoleOutput.java]: generic type T don't let message be primitive )

File: src/main/java/com/acme/dbo/txlog/ConsoleOutput.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.acme.dbo.txlog;
+
+public class ConsoleOutput {
```
Comment (eugene-krivosheyev) [inline src/main/java/com/acme/dbo/txlog/ConsoleOutput.java]: Why separate class for this responsibility?

File: src/main/java/com/acme/dbo/txlog/Facade.java [hunk 3]
```diff
@@ -1,11 +1,13 @@
 package com.acme.dbo.txlog;
 
+import static com.acme.dbo.txlog.ConsoleOutput.sysOut;
+
 public class Facade {
     public static void log(int message) {
-        System.out.println("primitive: " + message);
+        sysOut(message);
```
Comment (eugene-krivosheyev) [inline src/main/java/com/acme/dbo/txlog/Facade.java]: ok if you want make reusable part of output. What about part of decorating (prefixing) message?

**Review by eugene-krivosheyev — CHANGES_REQUESTED** [review]

File: src/main/java/com/acme/dbo/txlog/Facade.java (hunk 3 above)
Comment (AlexSamarkin) [reply src/main/java/com/acme/dbo/txlog/Facade.java]: oh, added decorator

File: src/main/java/com/acme/dbo/txlog/ConsoleOutput.java (hunk 2 above)
Comment (AlexSamarkin) [reply src/main/java/com/acme/dbo/txlog/ConsoleOutput.java]: I think it's the right way - we separate decorator logic to it's own class

File: src/main/java/com/acme/dbo/txlog/ConsoleOutput.java (hunk 1 above)
Comment (AlexSamarkin) [reply src/main/java/com/acme/dbo/txlog/ConsoleOutput.java]: yes, got it already, fixed)

**Pull request closed without merging**